src/main/resources/avro/alleleAnnotationmethods.avdl

```diff
@@ -0,0 +1,274 @@
+@namespace("org.ga4gh.methods")
+protocol AlleleAnnotationMethods {
+
+/*This protocol defines methods use to mine pre-calculated annotation sets.*/
+
+import idl "methods.avdl";
+import idl "variants.avdl";
+import idl "alleleAnnotations.avdl";
+import idl "sequenceAnnotations.avdl";
+import idl "ontologies.avdl";
+
+/*
+variantannotations/search returns annotation for the alleles of classic/linear variants
+alleleannotations/search returns annotation for graph alleles
+
+Both allow the mining of allele-specific annotations by either a region or a set of genomic
+features. Where a region is supplied annotation of all alleles vs all features in the region
+is returned. Where a set of features is supplied, only annotations against these features
+(matching on featuretype and id) are returned and other overlapping features are ignored.
+
+annotationsets/search returns information on reference data and software versions used in
+calculating the annotation. It is essential this information is exhaustive.
+*/
+
+/******************  /variantannotations/search  *********************/
+/** This request maps to the body of `POST /variantannotations/search` as JSON. */
+record SearchVariantAnnotationsRequest {
+  /** Required. The ID of the annotation set to search over. */
+  string annotationSetId;
+
+  /**
+  Only return annotation for variants which have exactly this name
+  (case-sensitive, exact match).
+  */
+  union { null, string } name = null;
+
+  /**
+  Only return variants with reference alleles on the reference with this name.
+  One of this field or `referenceId` or `features` is required.
+  (case-sensitive, exact match)
+  */
+  union { null, string } referenceName = null;
+
+  /**
+  Only return variants with reference alleles on the reference with this ID. One
+  of this field or `referenceName` or `features` is required.
+  */
+  union { null, string } referenceId = null;
+
+  /**
+  Required if referenceName or referenceId supplied.
```

Comment:
The required start/end was copied from master variants/search as an attempt at consistency. I see the reads/search has optional start and end which does make more sense.

The new refVar-linear branch, which is presumably soon to be master,  now searches variants by segment and reads by referenceId and optional start/end. @adamnovak - will reads/search be updated to take segments, meaning this is the standard variant annotations should also use?